internal/integration/integration_test.go

```diff
@@ -53,316 +69,323 @@ func TestIntegration(t *testing.T) {
 		OnlyLocalProvenance: false,
 	}
 
-	keys := util.GenerateExposureKeys(3, -1, false)
+	for _, tc := range cases {
 
-	// Publish 3 keys
-	payload := &verifyapi.Publish{
-		Keys:              keys,
-		HealthAuthorityID: "com.example.app",
-	}
-	jwtCfg.ExposureKeys = keys
-	verification, salt := testutil.IssueJWT(t, jwtCfg)
-	payload.VerificationPayload = verification
-	payload.HMACKey = salt
-	var revisionToken string
-	if resp, err := client.PublishKeys(payload); err != nil {
-		t.Fatal(err)
-	} else {
-		if resp.RevisionToken == "" {
-			t.Fatal("empty revision token")
-		}
-		revisionToken = resp.RevisionToken
-		t.Logf("response: %+v", resp)
-	}
+		keys := util.GenerateExposureKeys(3, -1, false)
 
-	// Assert there are 3 exposures in the database
-	{
-		exposures, err := getExposures(db, criteria)
-		if err != nil {
-			t.Fatal(err)
+		// Publish 3 keys
+		payload := &verifyapi.Publish{
+			Keys:              keys,
+			HealthAuthorityID: "com.example.app",
 		}
-		if got, want := len(exposures), 3; got != want {
-			t.Fatalf("expected %v to be %v", got, want)
+		jwtCfg.ExposureKeys = keys
+		jwtCfg.JWTWarp = tc.JWTWrap
+		verification, salt := testutil.IssueJWT(t, jwtCfg)
+		payload.VerificationPayload = verification
+		payload.HMACKey = salt
+		var revisionToken string
+		if resp, err := client.PublishKeys(payload); err != nil {
```

Comment:
This doesn't cover the case where error is expected but `PublishKeys` return nil

Reply:
done